packages/angular/src/generators/add-linting/lib/create-eslint-configuration.ts

```diff
@@ -21,9 +21,24 @@ export function createEsLintConfiguration(
           'plugin:@nrwl/nx/angular',
           'plugin:@angular-eslint/template/process-inline-templates',
         ],
-        parserOptions: {
-          project: [`${options.projectRoot}/tsconfig.*?.json`],
-        },
+        /**
+         * NOTE: We no longer set parserOptions.project by default when creating new projects.
+         *
+         * We have observed that users rarely add rules requiring type-checking to their Nx workspaces, and therefore
+         * do not actually need the capabilites which parserOptions.project provides. When specifying parserOptions.project,
+         * typescript-eslint needs to create full TypeScript Programs for you. When omitting it, it can perform a simple
+         * parse (and AST tranformation) of the source files it encounters during a lint run, which is much faster and much
+         * less memory intensive.
+         *
+         * In the rare case that users attempt to add rules requiring type-checking to their setup later on (and haven't set
+         * parserOptions.project), the executor will attempt to look for the particular error typescript-eslint gives you
+         * and provide feedback to the user.
+         */
+        parserOptions: !options.setParserOptionsProject
```

Comment:
I would invert the order here, instead of using negation. It's easier to reason about:
```typescript
parserOptions: options.setParserOptionsProject
  ? { project: [`${options.projectRoot}/tsconfig.*?.json`] }
  : undefined
```

Reply:
The reason for the order was to emphasise the default and common case. If there is strong support for changing it then I will be my subjective view is that that helps with clarity

Reply:
Ok, makes sense